### PR TITLE
[REF] Even less variable variables

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Further cleanup step in membership status calc

Before
----------------------------------------
ug variable variables

After
----------------------------------------
Less ug

Technical Details
----------------------------------------
@demeritcowboy 

Comments
----------------------------------------
See https://github.com/civicrm/civicrm-core/pull/18051 for relevant tests

### CODE REVIEW

File: CRM/Member/BAO/MembershipStatus.php [hunk 1]
```diff
@@ -263,21 +251,27 @@ public static function getMembershipStatusByDate(
...
-            ${$dat . 'Date'}
+          if (($membershipStatus->{$eve . '_event'} === $dat . '_date') &&
+            $date
           ) {
             if ($membershipStatus->{$eve . '_event_adjust_unit'} &&
               $membershipStatus->{$eve . '_event_adjust_interval'}
             ) {
-              $month = date('m', strtotime(${$dat . 'Date'}));
-              $day = date('d', strtotime(${$dat . 'Date'}));
-              $year = date('Y', strtotime(${$dat . 'Date'}));
```
Comment (totten) [inline CRM/Member/BAO/MembershipStatus.php:280]: Ooph, huzzah, remove that...

Comment (eileenmcnaughton) [reply CRM/Member/BAO/MembershipStatus.php:280]: Hey !!- that only just went in - you should see what the prior version looked like!

Comment (eileenmcnaughton) [reply CRM/Member/BAO/MembershipStatus.php:280]: Try this ....

```
   foreach ($dates as $dat) {
      if (${$dat . 'Date'} && ${$dat . 'Date'} != "null") {
        ${$dat . 'Date'} = CRM_Utils_Date::customFormat(${$dat . 'Date'}, '%Y%m%d');

        ${$dat . 'Year'} = substr(${$dat . 'Date'}, 0, 4);

        ${$dat . 'Month'} = substr(${$dat . 'Date'}, 4, 2);

        ${$dat . 'Day'} = substr(${$dat . 'Date'}, 6, 2);
      }
      else {
        ${$dat . 'Date'} = '';
      }
    }
```

**Comment by demeritcowboy** [comment]
Good work. It's getting there.

dollarsigns

**Pull request merged**